cloudevents/working-drafts/dds-format.md

```diff
@@ -0,0 +1,209 @@
+attributes are expressed using data types defined in the [Object Management
+Group (OMG)][omg] [Interface Definition Language (IDL) Specification][idl-spec].
+
+The [OMG DDS Specification][dds-spec] is closely related to the IDL specification as
+messages transmitted over the DDS protocol are defined via the IDL type system.
+These messages are sent over the wire using the Common Data Representation (CDR)
+serialization, as defined in the OMG [Real-Time Publish Subscribe (RTPS)][rtps]
+specification.
+
+It is noted here that the DDS Event Format is coupled to the DDS Protocol Binding
+for CloudEvents. CloudEvents defined in the DDS Event Format can only be
+decoded by a message receiver that is using the DDS protocol.  
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Attributes](#2-attributes)
+3. [Data](#3-data)
+4. [Transport](#4-transport)
+5. [Batch Format](#5-batch-format)
+6. [Examples](#6-examples)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how the
+elements defined in the CloudEvents specification are represented using a DDS
+schema.
+
+The [Attributes](#2-attributes) section describes the naming conventions and
+data type mappings for CloudEvent attributes for use as DDS message
+properties.
+
+The [Data](#3-data) section describes how the event payload is carried.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Content-Type
+
+There is no official IANA *media-type* designation for DDS, as such this
+specification uses 'application/dds' to identify such content.
+
+### 1.3 DDS Schema  
+
+DDS encoded CloudEvent messages MUST adhere to the format defined in the [DDS Schema][dds-schema].
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are represented in the [DDS schema][dds-schema].
+
+## 2.1 Type System
+
+The CloudEvents type system is mapped to IDL datatypes as follows :
+
+| CloudEvents   | DDS |
+| ------------- | ---------------------------------------------------------------------- |
+| Boolean       | [boolean][dds-schema] |
+| Integer       | [int32][dds-schema] |
+| String        | [string][dds-schema] |
+| Binary        | [bytes][dds-schema] |
+| URI           | [string][dds-schema] following [RFC 3986 §4.3][rfc3986-section43]|
+| URI-reference | [string][dds-schema] following [RFC 3986 §4.1][rfc3986-section41] |
+| Timestamp     | [Timestamp][dds-schema]  |
+
+## 2.3 REQUIRED Attributes
+
+REQUIRED attributes are represented explicitly as fields in the [Event type definition][dds-schema].
+
+## 2.4 OPTIONAL Attributes & Extensions
+
+At this time OPTIONAL and extension attributes are not supported in the DDS format.  
```

Comment:
That clause is in conflict with the base spec.

Reply:
@clemensv as always, thank you for this thoughtful review.

@JemDay had also remarked earlier about the interoperability concerns with DDS.

I agree with both of you. I actually ran a few simple experiments to see what would happen if a message that DDS did not know how to interpret was sent to a DDS subscriber. The DDS subscriber simply drops the message with a serialization error. And I believe that this is dictated by the DDS specification. So I believe this submission is at its termination point.

I was not aware that RTI had a DDS patent though. We always have to ensure that DDS is marked as an OMG specification for any public distribution. @clemensv would you send me a link to the patent if you have it?

Reply:
Please see response above

Reply:
@protimabanerjee -  

> would happen if a message that DDS did not know how to interpret was sent to a DDS subscriber

I'm a little confused by this as I would expect from a DDS perspective you're exchanging CE's, so the applications would produce, and consume CE's as defined in your format specification.